optaplanner-core/src/main/java/org/optaplanner/core/impl/score/constraint/DefaultConstraintMatchTotal.java

```diff
@@ -85,25 +86,36 @@ public Score getScore() {
     // ************************************************************************
 
     public ConstraintMatch addConstraintMatch(List<Object> justificationList, Score score) {
-        this.score = this.score.add(score);
-        ConstraintMatch constraintMatch = new ConstraintMatch(constraintPackage, constraintName,
-                justificationList, score);
-        boolean added = constraintMatchSet.add(constraintMatch);
-        if (!added) {
-            throw new IllegalStateException("The constraintMatchTotal (" + this
-                    + ") could not add constraintMatch (" + constraintMatch
-                    + ") to its constraintMatchSet (" + constraintMatchSet + ").");
```

Comment:
I've seen many score corruptions that were detected due to this fail fast exception. If this is removed, we're going to have more "ghosts in the machine" (= burn users).